plugins/content-manager/src/main/java/com/wazuh/contentmanager/cti/console/CtiConsole.java

```diff
@@ -108,19 +109,19 @@ public void onTokenChanged(Token t) {
 
     /**
      * Starts a periodic task to obtain a permanent token from the CTI Console.
-     * @param interval the period between successive executions.
+     * @param s subscription details, including the period between successive executions.
      */
-    private void getToken(int interval/* TODO sub details */) {
-        Runnable getTokenTask = () -> this.authService.getToken("client_id", "polling");
-        this.getTokenTaskFuture = this.executor.scheduleAtFixedRate(getTokenTask, interval, interval, TimeUnit.SECONDS);
+    private void getToken(Subscription s) {
+        Runnable getTokenTask = () -> this.authService.getToken(s);
+        this.getTokenTaskFuture = this.executor.scheduleAtFixedRate(getTokenTask, s.getInterval(), s.getInterval(), TimeUnit.SECONDS);
     }
 
     /**
      * Triggers the mechanism to obtain a permanent token from the CTI Console.
      * This method is meant to be called by the Rest handler.
      */
-    public void onPostSubscriptionRequest (/* TODO sub details */) {
-        this.getToken(5);
+    public void onPostSubscriptionRequest (Subscription s) {
+        this.getToken(s);
```

Comment:
Rename `s` to `subscription`
```suggestion
     * @param s subscription details, including the period between successive executions.
    */
    private void getToken(Subscription subscription) {
        Runnable getTokenTask = () -> this.authService.getToken(subscription);
        this.getTokenTaskFuture = this.executor.scheduleAtFixedRate(getTokenTask, subscription.getInterval(), subscription.getInterval(), TimeUnit.SECONDS);
}

/**
    * Triggers the mechanism to obtain a permanent token from the CTI Console.
    * This method is meant to be called by the Rest handler.
    */
    public void onPostSubscriptionRequest (Subscription subscription) {
        this.getToken(subscription);
```